ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -302,33 +322,55 @@ export const PageProviderDisplayBehavior = [
     },
 
     deselectItem(item) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllChecked) {
         this.$.list.deselectItem(item);
         this._updateFlags();
       }
     },
 
     deselectIndex(index) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllChecked) {
         this.$.list.deselectIndex(index);
         this._updateFlags();
       }
     },
 
     selectAll() {
-      this.items.forEach(
-        function(item) {
-          this.selectItem(item);
-        }.bind(this.$.list),
-      );
-      this._updateFlags();
+      if (this.selectionEnabled && this.selectAllEnabled) {
+        this._isSelectAllChecked = true;
+        // select the visible first (the others can be deferred)
+        const start = Math.max(0, this.$.list.firstVisibleIndex - 15);
+        const end = Math.min(this.items.length, this.$.list.lastVisibleIndex + 15);
```

Comment:
Why 15?

Reply:
Why not? 😂 Well, in fact the goal was to have 20 since the default page size is 40, but that means we would end up selecting too many items that were outside of the current page being displayed. It is more an empirical value 😬